### PR TITLE
WIP feat: manifest.json and running server in devmode

### PR DESCRIPTION
**Pull request by FKobus**
Work in progress

### CODE REVIEW

File: src/devServer/devServer.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export function createExpressApp(ctx: Context, props: IHTTPServerProps, extra?:
       extra.openProps.target = extra.openProps.target || `http://localhost:${props.port}`;
       open(extra.openProps.target, extra.openProps);
     }
-
+    ctx.log.clearLine();
```
Comment (FKobus) [inline src/devServer/devServer.ts]: most important line in this PR

File: src/production/bundleProd.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { createProductionContext } from './ProductionContext';
 import { Engine } from './engine';
 
 export async function bundleProd(ctx: Context): Promise<IRunResponse> {
+  ctx.log.startStreaming();
```
Comment (FKobus) [inline src/production/bundleProd.ts]: most important line of code in this PR

File: src/production/phases/BundlePhase.ts [hunk 3]
```diff
@@ -31,19 +31,19 @@ export async function BundlePhase(productionContext: IProductionContext) {
       router.generateSplitBundles(productionContext.splitEntries.entries);
     }
     await ctx.ict.resolve();
-    const bundles = await router.writeBundles();
+
+    const { bundles, manifest, onComplete } = await createRunResponse(ctx, router);
```
Comment (FKobus) [inline src/production/phases/BundlePhase.ts]: not sure about this... 🤔

File: src/devServer/server.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+import { BundleType, IBundleWriteResponse } from '../bundle/bundle';
+import { Context } from '../core/context';
+
+export interface IServerProcess {
+  start: (cliArgs: Array<string>) => ChildProcess;
+  stop: () => void;
+}
+
+export const createServerProcess = (ctx: Context, bundles: Array<IBundleWriteResponse>): IServerProcess => {
```
Comment (nchanged) [inline src/devServer/server.ts]: loving it ;-)

File: src/core/runResponse.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { IBundleRouter } from '../bundle/bundleRouter';
+import { createServerProcess, IServerProcess } from '../devServer/server';
+import { createServerEntry } from '../production/module/ServerEntry';
+import { Context } from './context';
+
+export const createRunResponse = async (ctx: Context, router: IBundleRouter) => {
+  const bundles = await router.writeBundles();
```
Comment (nchanged) [inline src/core/runResponse.ts]: if it's `createRunResponse` we shouldn't put writeBundles here

File: src/output/distWriter.ts [hunk 6]
```diff
@@ -76,7 +76,7 @@ export function distWriter(props: IOuputParserProps): DistWriter {
       if (userString.indexOf('$name') > -1) userString = userString.replace(/\$name/g, options.fileName);
       const absPath = ensureAbsolutePath(userString, root);
       const relativePath = ensureFuseBoxPath(path.relative(root, absPath));
-      const browserPath = options.publicPath ? options.publicPath + relativePath : relativePath;
+      const browserPath = options.publicPath ? path.join(options.publicPath, relativePath) : relativePath;
```
Comment (nchanged) [inline src/output/distWriter.ts]: It's on purpose with `+`
Now imagine windows users, they will get quite a surprise there in their index. On top of that, sometimes users would like to see `.` which path removes

File: src/production/phases/BundlePhase.ts (hunk 3 above)
Comment (nchanged) [reply src/production/phases/BundlePhase.ts]: me neither... Writing shouldn't be there

File: src/output/distWriter.ts (hunk 6 above)
Comment (FKobus) [reply src/output/distWriter.ts]: Then I'll update the code with more verbosity to add a forward slash if needed. I don't see why you would keep the `.` In browserpath tho

File: src/core/runResponse.ts (hunk 5 above)
Comment (FKobus) [reply src/core/runResponse.ts]: refactored

**Pull request merged**